Update contributing docs with pointers for legacy checkpoint testing (#8892)
PyTorchLightning/pytorch-lightning: .github/CONTRIBUTING.md

python -m pip install pre-commit
```

> Additionally, for testing back compatibility with legacy versions of Pytorch Lightning, you also need to download all saved version-checkpoints from the public AWS storage. Run the following script to get all saved version-checkpoints:

Reviewer: ```suggestion
Additionally, for testing backward compatibility with older versions of Pytorch Lightning, you also need to download all saved version-checkpoints from the public AWS storage. Run the following script to get all saved version-checkpoints:
```

> Additionally, for testing backward compatibility with older versions of Pytorch Lightning, you also need to download all saved version-checkpoints from the public AWS storage. Run the following script to get all saved version-checkpoints:

Reviewer 2: ```suggestion
Additionally, for testing backward compatibility with older versions of PyTorch Lightning, you also need to download all saved version-checkpoints from the public AWS storage. Run the following script to get all saved version-checkpoints:
```

Reviewer 2: I would like to also suggest to formulate it as optional. I believe to get started with contributing this is *absolutely not necessary*.

Reviewer: @awaelchli - in that case, this should be reflected with `make test` too: https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md#developments-scripts 

or `make test` should not run legacy checkpoint tests

```bash
cd legacy

> wget https://pl-public-data.s3.amazonaws.com/legacy/checkpoints.zip

Reviewer: might want to clarify which directory these must be downloaded to

Author: yep. I can update this like 
```bash
wget https://pl-public-data.s3.amazonaws.com/legacy/checkpoints.zip -P legacy/
unzip -o  legacy/checkpoints.zip -d legacy/
```

unzip -o checkpoints.zip
```

> Note: These checkpoints are generated to set baselines for maintaining back-compatibility with legacy versions of Pytorch Lightning. Details of checkpoints for back-compatibility can be found [here](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/legacy/README.md).

Reviewer: ```suggestion
Note: These checkpoints are generated to set baselines for maintaining backward compatibility with legacy versions of Pytorch Lightning. Details of checkpoints for back-compatibility can be found [here](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/legacy/README.md).
```

> Note: These checkpoints are generated to set baselines for maintaining backward compatibility with legacy versions of Pytorch Lightning. Details of checkpoints for back-compatibility can be found [here](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/legacy/README.md).

Reviewer 2: ```suggestion
Note: These checkpoints are generated to set baselines for maintaining backward compatibility with legacy versions of PyTorch Lightning. Details of checkpoints for back-compatibility can be found [here](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/legacy/README.md).
```
